rule/file-header.go

```diff
@@ -51,25 +51,34 @@ type lintFileHeader struct {
 	onFailure func(lint.Failure)
 }
 
+var (
+	multiRegexp  = regexp.MustCompile("^/\\*")
+	singleRegexp = regexp.MustCompile("^//")
+)
+
 func (w lintFileHeader) Visit(_ ast.Node) ast.Visitor {
-	g := w.fileAst.Comments[0]
 	failure := lint.Failure{
 		Node:       w.fileAst,
 		Confidence: 1,
 		Failure:    "the file doesn't have an appropriate header",
 	}
+
```

Comment:
We can take advantage of this PR to remove the unnecessary call to Visit. All the logic of this rule can be implemented without visiting the file AST. That is, it is possible to access and check the comments of the file at the `Apply` method.

Reply:
Done